python/lsst/daf/butler/_butler.py

```diff
@@ -1357,3 +1363,275 @@ def registry(self) -> Registry:
         will be replaced by equivalent `Butler` methods.
         """
         raise NotImplementedError()
+
+    @abstractmethod
+    def query(self) -> AbstractContextManager[Query]:
```

Comment:
Can we make this private for now, too?  While I'm not super worried about this method changing in backwards-compatible ways (it might get additional arguments, but they could be default), I do think we will want to tinker with the `Query` class APIs further as we actually implement more kinds of chaining, and it's easier to put the wall here than on every single method there.

Reply:
Keep it in with the underscores and when I make the release notes I can decide what to do with it.